improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,563 @@
+
+    def check_threshold_coordinate_properties(self, cube_name, threshold_coord):
+        """Checks threshold coordinate properties are correct and consistent with
+        cube name"""
+        threshold_var_name = threshold_coord.var_name
+        if threshold_var_name != "threshold":
+            self.errors.append(
+                f"Threshold coord {threshold_coord.name()} does not have "
+                "var_name='threshold'"
+            )
+
+        self.relative_to_threshold = threshold_coord.attributes[
+            "spp__relative_to_threshold"
+        ]
+
+        if self.relative_to_threshold in ("greater_than", "greater_than_or_equal_to"):
+            threshold_attribute = "above"
+        elif self.relative_to_threshold in ("less_than", "less_than_or_equal_to"):
+            threshold_attribute = "below"
+        elif self.relative_to_threshold == "between_thresholds":
+            threshold_attribute = "between"
+            self.warnings.append("Between thresholds data are not fully supported")
+        else:
+            threshold_attribute = None
+            self.errors.append(
+                f"spp__relative_to_threshold attribute '{self.relative_to_threshold}' "
+                "is not in permitted value set"
+            )
+
+        if threshold_attribute and threshold_attribute not in cube_name:
+            self.errors.append(
+                f"Cube name '{cube_name}' is not consistent with "
+                f"spp__relative_to_threshold attribute '{self.relative_to_threshold}'"
+            )
+
+    def check_cell_methods(self, cube):
+        """Checks cell methods are permitted and correct"""
+        cell_methods = cube.cell_methods
+        for cm in cell_methods:
+            if cm.method in COMP_CM_METHODS:
+                self.methods += f" {cm.method} over {cm.coord_names[0]}"
+                if self.field_type == self.PROB:
+                    if not cm.comments or cm.comments[0] != f"of {self.diagnostic}":
+                        self.errors.append(
+                            f"Cell method {cm} on probability data should have comment "
+                            f"'of {self.diagnostic}'"
+                        )
+                # check point and bounds on method coordinate
+                if "time" in cm.coord_names:
+                    if cube.coord("time").bounds is None:
+                        self.errors.append(f"Cube of{self.methods} has no time bounds")
+                    else:
+                        upper_bounds = [
+                            bounds[1] for bounds in cube.coord("time").bounds
+                        ]
+                        if not all(
+                            [
+                                point == bounds
+                                for point, bounds in zip(
+                                    cube.coord("time").points, upper_bounds
+                                )
+                            ]
+                        ):
```

Comment:
The time-zone cube is correct. The problem is the testing here which, by using list-comprehension, was assuming the time coord would always be 1D. The following code works for n-dim time coords, and is also much shorter.
```suggestion
                        upper_bounds = cube.coord("time").bounds[..., 1]
                        if not (cube.coord("time").points == upper_bounds).all():
 ```
PS. Iris 2 cannot print 2D time coords, but Iris 3 can, although you still get an error if you try to loop over the "cells" of a 2D time coord.